### ISSUE TITLE
Will function Get() of IntegrationPoint object provide access to both integration point and dimension?

### ISSUE THREAD

**Issue by AIBSCT80** [mfem/mfem#4365]
Hi, I wrote a small piece of code for setting coefficient in my pdes:

```c++
double MyCoefficientOmega :: Eval(ElementTransformation &T, const IntegrationPoint &ip)
{
   double omega=20;
   
   int dim;
   real_t p[3];
   ip.Get(p, dim);


   cout<<"dim:" << dim<< endl;

   double dist=0;
   for(int i=0; i<dim; i++)
   {
      dist+=p[dim]*p[dim];
   }
   dist=sqrt(dist);

   if(dist<=0.1)return 0.8*omega;
   else return 1.0*omega;
   
}
```

But I always get dim:0.  Is there anything wrong in my calling to `ip.Get` ? 

Many thanks.

**Comment by v-dobrev** [comment]
The parameter `dim` of `Get` in an input parameter, not an output. In the context of your function you can get the reference dimension from `T`: `T.GetDimension()` -- this is the dimension of the reference space and that's the dimension of the `IntegrationPoint` `ip`. The dimension of the physical space is given by `T.GetSpaceDim()` -- this is the dimension of physical points  as returned by calls to `T.Transform(ip, x)` where `x` is a `Vector`. Basically, `T` maps points from `T.GetDimension()`-dimensional space to `T.GetSpaceDim()`-dimensional space.

**Comment by AIBSCT80** [comment]
Hi @v-dobrev 
many thanks for your help.  I tried to correct my code accordingly, basically, I want to check each of the integration points in the real physical space in order to assign a coefficient according to the real physical position.  

Now, I first get point x in physical space, then assign the coefficient to x according to their physical point:


double MyCoefficientOmega :: Eval(ElementTransformation &T, const IntegrationPoint &ip)
{

   double omega=2.0*M_PI*10;
   
   int dim=T.GetSpaceDim();
   Vector x;
   T.Transform(ip, x);

   double dist=0;
   for(int i=0; i<dim; i++)
   {
      dist+=x[dim]*x[dim];
   }
   dist=sqrt(dist);

   if(dist<=0.1)return 0.8*omega;
   else return 1.0*omega;
   
}

I hope I did it correctly this time.  Many many thanks for your time.
